pkg/clustersync/clusterSync.go

```diff
@@ -175,6 +175,18 @@ func processClusterUpsert(ctx context.Context, obj interface{}) {
 	// Upsert (attempt insert, update on failure)
 	dao.UpsertCluster(ctx, resource)
 
+	// Confirm deleted cluster data doesn't remain in db:
+	clusterRemaining, err := confirmDelete(ctx)
+	if err != nil {
+		klog.Warning("Error confirming cluster deletion", err.Error())
+	} else if len(clusterRemaining) > 0 {
+		for _, cluster := range clusterRemaining {
+			dao.DeleteClusterAndResources(ctx, cluster, false)
+		}
+	} else {
+		klog.V(5).Infof("Managed Cluster data deleted successfully.")
+	}
```

Comment:
This gets invoked too often.  We only need to check once, when starting the indexer.
Also, this won't get executed if there isn't any managed cluster.

Reply:
okay moved to earlier stage. so we would need to check the client and the search-addon - if nothing there than no need to query db.